torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py

```diff
@@ -168,83 +166,45 @@ def _annotate_linear(
     quantization_config: Optional[QuantizationConfig],
     filter_fn: Optional[Callable[[Node], bool]] = None,
 ) -> Optional[List[List[Node]]]:
-    module_partitions = get_source_partitions(
-        gm.graph, [torch.nn.Linear, torch.nn.functional.linear], filter_fn
-    )
+    annotated_partitions = []
     input_act_qspec = get_input_act_qspec(quantization_config)
     output_act_qspec = get_output_act_qspec(quantization_config)
     weight_qspec = get_weight_qspec(quantization_config)
     bias_qspec = get_bias_qspec(quantization_config)
-    annotated_partitions = []
-    for partitions in module_partitions.values():
-        for p in partitions:
-            annotated_partitions.append(p.nodes)
-            act_nodes = [
-                n for n in p.input_nodes if not _node_only_used_for_sym_size(n, p.nodes)
-            ]
-            if len(act_nodes) > 1:
-                raise ValueError(
-                    f"Multiple activation nodes found for partition {p} {act_nodes}"
-                )
-            if len(act_nodes) == 0:
-                raise ValueError(f"No activation node found for partition {p}")
-            act_node = act_nodes[0]
-            output_node = p.output_nodes[0]
-            weight_node = None
-            bias_node = None
-            for node in p.params:
-                weight_or_bias = getattr(gm, node.target)  # type: ignore[arg-type]
-                if weight_or_bias.ndim == 2:  # type: ignore[attr-defined]
-                    weight_node = node
-                if weight_or_bias.ndim == 1:  # type: ignore[attr-defined]
-                    bias_node = node
-            if weight_node is None:
-                raise ValueError("No weight found in Linear pattern")
-            # find use of act node within the matched pattern
-            act_use_node = None
-            # When doing tracing with dynamic shape, we end up with sym_size nodes
-            # This nodes do not need quantization, so skip those.
-            # We can also have quant workflow throw exception when sym_size nodes
-            # are annotated.
-            # This is not specific to linear, so in future diffs we should streamline
-            # this.
-            weight_node_users = list(
-                filter((lambda x: (_is_sym_size_node(x) is False)), weight_node.users)
+    for node in gm.graph.nodes:
+        if node.op != "call_function" or node.target != torch.ops.aten.linear.default:
+            continue
+        if filter_fn and not filter_fn(node):
+            continue
+        annotated_partitions.append([node])
+        act_node = node.args[0]
+        weight_node = node.args[1]
+        bias_node = None
+        if len(node.args) < 2:
```

Comment:
is this enforced by the schema for aten.linear op itself?

Reply:
you cant have aten.linear without input + weight. https://fburl.com/code/knz9927g

Reply:
right, I mean we don't need to check for this since it's enforced by the schema?

Reply:
oh i see what you mean. yes dont need to.

Reply:
are you planning to remove this?

Reply:
oh sorry forgot